### PR TITLE
volume: Change owner of symlinks too

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This commit uses Lchown instead of Chown to change the owner of symlinks too.
It doesn't change any behaviour. However, it could avoid some confusions as the
symlinks are updated to the correct owner too.

Please consider the following Pod with a configMap:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: myconfigmap
  namespace: default
data:
  content: '"This is a ConfigMap"'
---
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  securityContext:
    fsGroup: 2000
  containers:
  - name: container1
    image: busybox
    command: ["sh"]
    args: ["-c", "sleep infinity"]
    volumeMounts:
    - name: config-volume
      mountPath: /etc/configmap/content
  volumes:
  - name: config-volume
    configMap:
      name: myconfigmap
      defaultMode: 0400
```

The current behaviour shows that the group owner is `0` for the symlink point to the real content file:
```
$ kubectl exec -it mypod -- /bin/sh -c 'ls -la /etc/configmap/content/content'
lrwxrwxrwx    1 0     0            14 Sep 18 15:15 /etc/configmap/content/content -> ..data/content
```

The behaviour after this PR shows as group owner `2000` as specified in `fsGroup`:
```
kubectl exec -it mypod -- /bin/sh -c 'ls -la /etc/configmap/content/content'
lrwxrwxrwx    1 0     2000            14 Sep 18 15:15 /etc/configmap/content/content -> ..data/content
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/volume/volume_linux.go [hunk 1]
```diff
@@ -82,32 +82,22 @@ func legacyOwnershipChange(mounter Mounter, fsGroup *int64) error {
...
-	if stat == nil {
-		klog.Errorf("Got nil stat_t for path %v while setting ownership of volume", filename)
-		return nil
-	}
-
-	err := os.Chown(filename, int(stat.Uid), int(*fsGroup))
-	if err != nil {
-		klog.Errorf("Chown failed on %v: %v", filename, err)
-	}
-
```
Comment (gnufied) [inline pkg/volume/volume_linux.go:110]: Isn't this changing the existing behaviour?

Comment (mauriciovasquezbernal) [reply pkg/volume/volume_linux.go:110]: Strictly speaking, yes. Now the group owner of the symlinks is updated too. Am I missing anything else?

**Comment by jingxu97** [comment]
could you add unit test? Thanks!

cc @msau42

**Comment by mauriciovasquezbernal** [comment]
@jingxu97 I added the unit test.

**Comment by mauriciovasquezbernal** [comment]
@gnufied @sjenning any chance you can give it a look?

**Comment by gnufied** [comment]
It looks good to me as well. But it would be nice to have a confirmation from @sjenning because he originally made the change that prevented permissions of symlinks from being changed. But at that time - we were calling plain` chown` which evaluates the symlink and now you are using `lchown`.

**Comment by joelsmith** [comment]
Nice that you were able to get rid of an extra stat syscall by using -1 for the UID!

/lgtm

**Pull request merged**